inject/src/main/java/io/micronaut/context/DefaultBeanContext.java

```diff
@@ -2214,6 +2214,18 @@ private void readAllBeanDefinitionClasses() {
         disabled.clear();
 
         initializeEventListeners();
+        if (containsBean(MessageSource.class)) {
```

Comment:
Is this necessary? Can't you inject `@Secondary` instances into the composite primary bean?

Reply:
Yeah maybe not necessary